lib/src/rules/noop_primitive_operations.dart

```diff
@@ -66,10 +70,21 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
   }
 
+  @override
+  void visitInterpolationExpression(InterpolationExpression node) {
+    _checkToStringInvocation(node.expression);
+  }
+
   @override
   void visitMethodInvocation(MethodInvocation node) {
     var type = node.realTarget?.staticType;
-    if (type == null) return;
+    if (type == null) {
+      // print(xxx.toString())
+      if (node.methodName.staticElement?.isDartCorePrint ?? false) {
```

Comment:
You don't need `?.` here - the extension already works on `Element?`.

Which itself might be a good lint? :-)

Reply:
Should it go into `invalid_null_aware_operator` ?

Reply:
https://github.com/dart-lang/sdk/pull/48980 is an attempt to cover this case.